lightning/src/chain/keysinterface.rs

```diff
@@ -423,12 +433,36 @@ pub trait KeysInterface {
 	/// this trait to parse the invoice and make sure they're signing what they expect, rather than
 	/// blindly signing the hash.
 	/// The hrp is ascii bytes, while the invoice data is base32.
-	fn sign_invoice(&self, hrp_bytes: &[u8], invoice_data: &[u5]) -> Result<RecoverableSignature, ()>;
+	///
+	/// If `invoice_type` is [`Invoice::Bolt11`], then this invoice must be signed by the node secret
+	/// provided by [`KeysInterface::get_node_secret`]. Else if it is [`Invoice::Phantom`], then this
+	/// invoice must be signed by the phantom node secret provided by
+	/// [`KeysInterface::get_phantom_secret`].
+	fn sign_invoice(&self, hrp_bytes: &[u8], invoice_data: &[u5], invoice_type: Invoice) -> Result<RecoverableSignature, ()>;
 
 	/// Get secret key material as bytes for use in encrypting and decrypting inbound payment data.
 	///
+	/// If the implementor of this trait supports [phantom node payments], then every node that is
+	/// intended to be included in the phantom invoice(s) route hints must return the same value from
+	/// this method.
+	///
 	/// This method must return the same value each time it is called.
+	///
+	/// [phantom node payments]: crate::ln::channelmanager::ChannelManager::get_phantom_scid
 	fn get_inbound_payment_key_material(&self) -> KeyMaterial;
+
+	/// Get the phantom node secret key (aka node_id or network_key) for use in receiving a [phantom
```

Comment:
I'm confused here with the "aka node_id or network_key" as the same interface also offers a `get_node_secret`. Just to be sure `phantom_secret` (shared) != `node_secret` (local).

Reply:
Yup, that's right. Matt suggested this change [here](https://github.com/lightningdevkit/rust-lightning/pull/1199#discussion_r787098937)